appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/impl/ConcurrentGaugeImpl.java

```diff
@@ -62,25 +63,53 @@
     /**
      * Last point for which clearOld() has been run
      */
-    private Instant lastInstant = Instant.now();
-    
+    private Instant lastInstant = getCurrentMinute();
+
+    /**
+     * A map containing all the max count values that the count of the gauge has
+     * been at in the last and current minutes. The key is the time in minute
+     * that the value changed, the value is the max value of the count at the
+     * moment before it changed
+     */
+    private final ConcurrentMap<Instant, Long> maxCounts
+            = new ConcurrentHashMap<>();
+
     /**
-     * A map containing all the values that the count of the gauge has been at in the last minute.
-     * The key is the time in seconds that the value changed, the value is the value of the count at the moment
-     * before it changed
+     * A map containing all the min count values that the count of the gauge has
+     * been at in the last and current minutes. The key is the time in minute
+     * that the value changed, the value is the min value of the count at the
+     * moment before it changed
      */
-    private Map<Instant, Long> lastCounts = new ConcurrentHashMap<>();
-    
+    private final ConcurrentMap<Instant, Long> minCounts
+            = new ConcurrentHashMap<>();
+
     /**
      * Increment the counter by one.
      */
     @Override
     public void inc() {
         clearOld();
-        if (count.longValue() > 0) {
-            lastCounts.put(Instant.now(), count.longValue());
-        }
-        count.increment();
+        Instant currentMin = getCurrentMinute();
+
+            count.increment();
+            long latestCount = count.longValue();
+            Long value = maxCounts.getOrDefault(currentMin, 0L);
+            if (latestCount > value) {
+                maxCounts.put(currentMin, latestCount);
+            }
```

Comment:
These needs to utilize update functions. Otherwise the moment you arrive at `maxCounts.put`, any of the following might have happened:
* latestCount is less than actual value of counter
* value no longer represents captured maximum
* other thread already put updated maximum, greater than latestCount